logback-classic/src/main/java/ch/qos/logback/classic/android/Log.java

```diff
@@ -0,0 +1,827 @@
+/**
+ * Logback: the reliable, generic, fast and flexible logging framework.
+ * Copyright (C) 1999-2013, QOS.ch. All rights reserved.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation.
+ */
+
+package ch.qos.logback.classic.android;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+
+/**
+ * Simple Android-like adapter of {@link Logger} interface.
+ * <p/>
+ * Any call like {@code Log.someMethod(arguments)} is equal to:
+ * <p/>
+ * <code>
+ * {@link #getLogger()}.someMethod(arguments);
+ * </code>
+ * <p/>
+ * <b>Note</b>:
+ * Do not forget to configure logback before use.
+ *
+ * @author NoveoGroup
+ * @author Pavel Stepanov
+ */
+public class Log {
+
+  private static final class CallerResolver extends SecurityManager {
+    public Class<?> getCaller(Class<?> aClass) {
+      boolean classFound = false;
+      Class[] classContext = CALLER_RESOLVER.getClassContext();
+      if (classContext == null) {
+        return null; // sometimes class context is null
+      }
+      for (Class callerClass : classContext) {
+        if (!classFound) {
+          if (callerClass == aClass) {
+            classFound = true;
+          }
+        } else {
+          if (callerClass != aClass) {
+            return callerClass;
+          }
+        }
+      }
+      return classContext[classContext.length - 1];
+    }
+  }
+
+  private static final CallerResolver CALLER_RESOLVER = new CallerResolver();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/tony19/logback-android/file/2349083805/issues/source?bid=3217169&fileBranchId=3217175#l60)